src/rules/shared.ts

```diff
@@ -170,7 +170,7 @@ export const replaceBeginEndBody = (
     selector.begin,
     selector.end
   )
-  if (!valueToReplace) return request
+  if (!valueToReplace) return
```

Comment:
Previously, we had no way of knowing if the request matched when the value replaced is the same previous value. While it does not change script, it still should be displayed as matched request in rule preview. 

Returning `undefined` when there's no match allows consumers to know when rule matched.